### PR TITLE
NIFI-9081 Updated nifi-framework-api tests to JUnit 5.

### PR DESCRIPTION
**Pull request by MikeThomsen**
Updated nifi-external tests to JUnit 5.
Updated nifi-stateless to use JUnit 5.
Updated nifi-api to use JUnit 5.

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-external/nifi-kafka-connect/nifi-kafka-connector-tests/src/test/java/org/apache/nifi/kafka/connect/StatelessNiFiSinkTaskIT.java [hunk 1]
```diff
@@ -121,27 +117,27 @@ public void testWrongOutputPort() {
...
         }
 
         try {
             sinkTask.put(Collections.singleton(record));
             sinkTask.flush(Collections.emptyMap());
-            Assert.fail("Expected RetriableException to be thrown");
+            fail("Expected RetriableException to be thrown");
         } catch (final RetriableException re) {
             // Expected
         }
```
Comment (exceptionfactory) [inline nifi-external/nifi-kafka-connect/nifi-kafka-connector-tests/src/test/java/org/apache/nifi/kafka/connect/StatelessNiFiSinkTaskIT.java:130]: This should be changed to use `assertThrows()`.

File: nifi-external/nifi-kafka-connect/nifi-kafka-connector-tests/src/test/java/org/apache/nifi/kafka/connect/StatelessNiFiSourceTaskIT.java [hunk 2]
```diff
@@ -133,24 +129,24 @@ public void testHeaders() throws InterruptedException {
...
         properties.put(StatelessNiFiSourceConnector.OUTPUT_PORT_NAME, "Another");
         sourceTask.start(properties);
 
         try {
             sourceTask.poll();
-            Assert.fail("Expected RetriableException to be thrown");
+            fail("Expected RetriableException to be thrown");
         } catch (final RetriableException re) {
             // Expected
         }
```
Comment (exceptionfactory) [inline nifi-external/nifi-kafka-connect/nifi-kafka-connector-tests/src/test/java/org/apache/nifi/kafka/connect/StatelessNiFiSourceTaskIT.java:145]: This should be changed to use `assertThrows()`.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
This pull request looks close to completion, thanks @MikeThomsen. Just noted to places where `fail()` inside a try-catch could be replaced with `assertThrows`, otherwise this looks good.

**Comment by MikeThomsen** [comment]
@exceptionfactory changes made.

**Comment by exceptionfactory** [comment]
Thanks @MikeThomsen, looks good! +1 Merging.

**Comment by exceptionfactory** [comment]
Merged in https://github.com/apache/nifi/commit/6a55c6ca447ea902e454daf29eb9d17720310219

**Pull request closed without merging**